File: cpp/cw_funkcje02.cpp
#include <iostream>
using namespace std;


void srednia(float suma, float iloscliczb){
    float srednia = suma / iloscliczb;
    cout <<"Œrednia wynosi: "<< srednia <<endl;
}



void med(float *tab, int iloscliczb){
    float med;
    if(iloscliczb % 2 ==0){
        med = (tab[iloscliczb/2 - 1]+tab[iloscliczb/2])/2;
    } else {
        med = tab[iloscliczb/2];
    }
    cout <<"Mediana wynosi: " << med << endl;
}


int main(int argc, char **argv)
{

	int t =0;
	int iloscliczb =0;
	int *tab = new int [iloscliczb];
	cout <<"Iloœæ liczb: ";
	cin >>iloscliczb;

	if(iloscliczb >20){
		cout <<"Maksymalna iloœæ ocen wynosi 20" <<endl;
		return 0;
}

	float tablica[iloscliczb];
	int suma = 0;
	for (int i = 0; i < iloscliczb; ++i)
    {
		cout <<"Podaj ocenê  "<<(i+1)<<" : ";
		cin >>tablica[i];
		if(tablica[i] <1 || tablica[i] >6){
			cout <<"Wprowadzono niepoprawn¹ ocenê"<<endl;
			return 0;
        }
		suma += tablica[i];
    }
	int v;
	int x;
	for(v=0;v<iloscliczb;v++)
    	{
			for(x=v+1;x<iloscliczb;x++)
    		{
    			if(tablica[v]>tablica[x])
    			{
    				t  =tablica[v];
    				tablica[v]=tablica[x];
    				tablica[x]=t;
    			}
    		}
    	}
    srednia(suma, iloscliczb);
    med(tablica, iloscliczb);


	return 0;
}

File: python/cw_funkcje02.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  cw_funkcje02.py
#  
#  Copyright 2020 Karol <Karol@DESKTOP-EDM589B>
#  
#  This program is free software; you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation; either version 2 of the License, or
#  (at your option) any later version.
#  
#  This program is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#  
#  You should have received a copy of the GNU General Public License
#  along with this program; if not, write to the Free Software
#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
#  MA 02110-1301, USA.
#  

import math
import statistics
def srednia(tablica, iloscliczb):
	srednia =sum(tablica)/iloscliczb
	print("Średnia wynosi: ", srednia)


def mediana(tablica):
	print("Mediana wynosi: ", statistics.median(tablica))



def main(args):
	iloscliczb = int(input("Ilość liczb: "))
	if iloscliczb >20:
		print("Maksymalna ilość ocen wynosi 20!")
		return(iloscliczb)

	tablica =[]
	for i in range(0,iloscliczb):
		f = int(input("Podaj liczbę: "))
		tablica.append(f)

		if f <=0 or f >6:
			print("Wprowadzono niepoprawną ocenę")
			return


	srednia(tablica, iloscliczb)
	mediana(tablica)

	return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
